tests/js/session-tests.js

```diff
@@ -545,7 +547,93 @@ module.exports = {
         const session = realm.syncSession;
 
         TestCase.assertEqual(session.config.error, config.sync.error);
-        session._simulateError(211, "ClientReset"); // 211 -> divering histories
+        session._simulateError(211, "Simulate Client Reset", "realm::sync::ProtocolError", false); // 211 -> diverging histories
+      });
+    });
+  },
+
+  testClientResetDiscardLocalFailed() {
+    if (!platformSupported) {
+      return;
+    }
+
+    // if client reset fails, the error handler is called
+    // and the two before/after handlers are not called
+    // we simulate the failure by error code 132
+    const partition = Utils.genPartition();
+    let creds = Realm.Credentials.anonymous();
+    let app = new Realm.App(appConfig);
+    return new Promise((resolve, reject) => {
+      return app.logIn(creds).then((user) => {
+        const config = getSyncConfiguration(user, partition);
+        config.sync.clientReset = {
+          mode: "discardLocal",
+          clientResetBefore: (localRealm) => {
+            reject("clientResetBefore");
+          },
+          clientResetAfter: (localRealm, remoteRealm) => {
+            reject("clientResetAfter");
+          },
+        };
+        config.sync.error = (sender, error) => {
+          resolve();
+        };
+
+        Realm.open(config).then((realm) => {
+          realm.syncSession._simulateError(132, "Simulate Client Reset", "realm::sync::ProtocolError", true); // 132 -> automatic client reset failed
+        });
+      });
+    });
+  },
+
+  testClientResetDiscardLocal() {
+    if (!platformSupported) {
+      return;
+    }
+
+    // (i)   using a client reset in "DiscardLocal" mode, a fresh copy
+    //       of the Realm will be downloaded (resync)
+    // (ii)  two callback will be called, while the sync error handler is not
+    // (iii) after the reset, the Realm can be used as before
+
+    let beforeCalled = false;
+    let afterCalled = false;
+
+    const partition = Utils.genPartition();
+    let creds = Realm.Credentials.anonymous();
+    let app = new Realm.App(appConfig);
+    return new Promise((resolve, reject) => {
+      return app.logIn(creds).then((user) => {
+        const config = getSyncConfiguration(user, partition);
+        config.sync.clientReset = {
+          mode: "discardLocal",
+          clientResetBefore: (localRealm) => {
+            beforeCalled = true;
+            TestCase.assertEqual(localRealm.objects("Dog").length, 1, "local");
+          },
+          clientResetAfter: (localRealm, remoteRealm) => {
+            afterCalled = true;
+            TestCase.assertEqual(localRealm.objects("Dog").length, 1, "local");
+            // TestCase.assertEqual(remoteRealm.objects("Dog").length, 1, "remote");
+            // TestCase.assertNotEqual(remoteRealm.path, localRealm.path);
```

Comment:
Since the most recent change, these paths will actually be the same. The `before` state is a an older frozen version of the same Realm and the `after` state is the live most up to date version after the reset has been committed.

Reply:
Got it; it doesn't make sense to compare the two paths.